config/src/config.rs

```diff
@@ -601,13 +604,44 @@ impl NodeConfig {
             .any(|network| RoleType::Validator == (&network.role).into())
     }
 
+    /// Returns true if the node config is for a validator. Otherwise returns false.
+    pub fn get_role(&self) -> RoleType {
+        if self.is_validator() {
+            RoleType::Validator
+        } else {
+            RoleType::FullNode
+        }
+    }
+
     /// Returns the validator network config for this node.
     pub fn get_validator_network_config(&self) -> Option<&NetworkConfig> {
         self.networks
             .iter()
             .filter(|network| RoleType::Validator == (&network.role).into())
             .last()
     }
+
+    /// Returns true if network_config is for an upstream network
+    pub fn is_upstream_network(&self, network_config: &NetworkConfig) -> bool {
+        self.state_sync
+            .upstream_peers
```

Comment:
nit: should we use here `get_upstream_peer_ids` function?

Reply:
not sure if those two functions are compatible?

Reply:
Seems like `network_config.network_peers.peers` is a `HashMap<String, NetworkPeerInfo>` so I don't think it's necessary to convert from string to PeerId before checking `.any(|peer_id|network_config.network_peers.peers.contains_key(peer_id))`

.any(|peer_id| network_config.network_peers.peers.contains_key(peer_id))